docs/auth/multiple-auth-modes.mdx

```diff
@@ -0,0 +1,159 @@
+---
+sidebar_position: 6
+description: "Learn how to configure and use multiple authentication modes in Hasura DDN."
+keywords:
+  - hasura authentication
+  - multiple auth modes
+  - jwt authentication
+  - webhook authentication
+  - noauth mode
+  - auth switching
+  - X-Hasura-Auth-Mode
+sidebar_label: Multiple Auth Modes
+---
+
+import Thumbnail from "@site/src/components/Thumbnail";
+
+# Multiple Authentication Modes
+
+## Introduction
+
+Hasura DDN supports configuring multiple authentication modes simultaneously, allowing you to use different
+authentication methods for different scenarios or clients. This feature enables you to have both JWT and webhook
+authentication enabled, or multiple configurations of the same authentication type with different settings.
+
+:::info AuthConfig v4
+
+Multiple authentication modes are available in `AuthConfig` **v4** and above.
+
+:::
+
+## How It Works
+
+When multiple authentication modes are configured:
+
+1. The primary mode specified in the `mode` field of the `AuthConfig` is used by default
+2. Alternative modes can be accessed by passing the `X-Hasura-Auth-Mode` header with the identifier of the desired
+   authentication mode
+3. Each alternative mode has its own complete authentication configuration
+
+The default mode is used when:
+
+- The `X-Hasura-Auth-Mode` header is not provided
+- The provided `X-Hasura-Auth-Mode` header does not match any of the alternative mode identifiers
+
+## Configuring Multiple Auth Modes
+
+To configure multiple authentication modes, you need to use `AuthConfig` v4, which introduces the `alternativeModes`
+field.
+
+### Example Configuration
+
+Here's an example of an `AuthConfig` with multiple authentication modes:
+
+```yaml
+kind: AuthConfig
+version: v4
+definition:
+  mode:
+    jwt:
+      key:
+        fixed:
+          algorithm: HS256
+          key:
+            valueFromEnv: JWT_KEY
+      tokenLocation:
+        type: BearerAuthorization
+      claimsConfig:
+        namespace:
+          claimsFormat: Json
+          location: "/https:~1~1hasura.io~1jwt~1claims"
+  alternativeModes:
+    - identifier: "user2"
+      config:
+        webhook:
+          url: "https://my-auth-service.example.com/validate-user2"
+          method: "POST"
+    - identifier: "webhook"
+      config:
+        webhook:
+          url: "https://my-auth-service.example.com/validate"
+          method: "POST"
+    - identifier: "jwt"
+      config:
+        jwt:
+          jwkUrl: "https://www.googleapis.com/service_accounts/v1/jwk/securetoken@system.gserviceaccount.com"
+          audience: "my-firebase-app"
+          issuer: "https://securetoken.google.com/my-firebase-app"
+          claims_namespace: "https://hasura.io/jwt/claims"
```

Comment:
Just checking if these are defined correctly? eg:
- webhook url is supposed to have `value` or `valueFromEnv`?
- Is it jwkUrl and not jwkFromUrl as per [here](paritosh-auth-config-v4.v3-docs-eny.pages.dev/auth/jwt/jwt-configuration#jwt-json-jwk_url)?
- etc.

Reply:
Thanks for catching this, I have fixed them and cross-checked with an actual project.